src/CMakeLists.txt

```
@@ -36,6 +38,7 @@ target_link_libraries(${PROJECT_NAME}
 
 # See https://github.com/boostorg/beast/issues/2661
 target_compile_definitions(${PROJECT_NAME} PRIVATE BOOST_ASIO_DISABLE_CONCEPTS)
+target_compile_definitions(${PROJECT_NAME} PUBLIC $<$<BOOL:{WIN32}>:BOOST_ALL_NO_LIB>)
```

This generator expression was missing a $ for the WIN32 variable, causing this to evaluate to true on all platforms since bool of "WIN32" is 1 (of course)

```suggestion
target_compile_definitions(${PROJECT_NAME} PUBLIC $<$<BOOL:${WIN32}>:BOOST_ALL_NO_LIB>)
```